lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -219,10 +220,18 @@ public Builder setDimensions(DimensionList dimensions) {
     public Builder setTimestamp(Instant timestamp) {
       int year = timestamp.atZone(ZoneOffset.UTC).getYear();
       if (year < 2000 || year > 3000) {
-        logger.warning(
-            "Order of magnitude of the timestamp seems off. "
-                + "The timestamp represents a time before the year 2000 or after the year 3000. "
-                + "Skipping setting timestamp, the current server time will be added upon ingestion.");
+        if (numberOfTimestampWarnings == 0) {
+          logger.warning(
+              "Order of magnitude of the timestamp seems off. "
+                  + "The timestamp represents a time before the year 2000 or after the year 3000. "
+                  + "Skipping setting timestamp, the current server time will be added upon ingestion. "
+                  + "Only every 1000th of these messages will be printed.");
+        }
+        numberOfTimestampWarnings++;
+        if (numberOfTimestampWarnings == 1000) {
```

Comment:
Please add a constant for that so both occurrences stay in sync if it was changed.
It would also make sense to print the Instant in question in the warning.

Reply:
I'm +1 on making this a constant, but what do you mean 'both occurrences'? I only see one here.

edit: I see in the text nevermind